src/protocol.rs

```diff
@@ -469,10 +469,10 @@ impl Payout {
             }
             (false, true) => {
                 updated.maker_amount = Amount::ZERO;
-                updated.taker_amount = taker_amount - (fee + maker_amount);
+                updated.taker_amount = taker_amount - fee + maker_amount;
```

Comment:
@luckysori : 
Note: this is a breaking change IF either party would receive 0.
I'm not sure we need to cover for this. Any thoughts?

Reply:
What I understand is:

- Before, we gave the party who deserved the full total amount much less than what they were supposed to. Basically all liquidation CETs are broken.
- With these changes we fix this critical (but highly unlikely to affect us) bug.
- If the production maker uses this code against and old taker, they will always fail during contract setup because they will disagree on the payouts for the one liquidation CET.

Basically, it is hard to make changes to the protocol without breaking old clients. This change is correct in `maia`, but blindly applying it to `itchysats` will make previous versions incompatible. We should really consider staying compatible with old versions i.e. only deprecating this behaviour for the next version of `taker`.

Reply:
I agree, how would you design backwards compatibility in this case?

Reply:
Depend on two versions on `maia` and differentiate between an old version of the collaborative settlement protocol and the  new one.

Apparently one can depend on multiple versions of the same crate using the [rename-dependency cargo feature](https://stackoverflow.com/questions/58739075/how-do-i-import-multiple-versions-of-the-same-crate).